addons/mail/static/src/models/message/message.js

```diff
@@ -442,6 +442,32 @@ function factory(dependencies) {
                 this.originThread.isModeratedByCurrentPartner
             );
         }
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsSubjectEqualThreadName() {
+            if (!this.subject || !this.originThreadName) {
+                return false;
+            }
+
+            const threadName = this.originThreadName.toLowerCase().trim();
+            let prefix = ['re:', 'fw:', 'fwd:'];
+            let cleanedSubject = this.subject.toLowerCase();
+            let startingSubject;
+            while (cleanedSubject !== startingSubject) {
+                startingSubject = cleanedSubject;
+                for (const pref of prefix) {
+                    if (cleanedSubject.startsWith(pref)) {
+                        cleanedSubject = cleanedSubject.replace(pref, '').trim();
+                        break;
+                    }
+                }
+                if (threadName === startingSubject) {
+                   return true;
+                }
+            }
```

Comment:
Don't forget to `return false;` at the end here, because without it the method will return `undefined`, which will keep the current value (possibly the default value) of the field `isSubjectEqualThreadName`. So if for example the subject or the thread name is updated, this will not update correctly to `false`.